pyshelf/search/sort_base.py

```diff
@@ -0,0 +1,17 @@
+class SortBase(object):
+    @classmethod
+    def get_sort_alias(cls, flag, aliases):
+        """
+            Determines if passed in flag is a SortFlag or an alias of a SortFlag.
+            Args:
+                flag(string): potential sort base alias.
+                aliases(Dict{list}): aliases for attributes.
+
+            Returns:
+                SortBase or None: returns None if invalid flag.
+        """
+        for key, val in aliases.iteritems():
+            if flag.upper() in val:
```

Comment:
This is not going to work in a generic sense.  It feels very confusing.  In other words if I passed in
```
ALIASES = {
    "BOB": [
        "JANE" 
     ]
}
```

It wouldn't work.

I think instead you should actually loop through the other values or structure it differently so it can be used as a simple map.

```
ALIASES = {
    "JANE": SortType.BOB,
    "ALICE": SortType.BOB
}
```
Then inside the base class do

```
return aliases.get(flag.upper())
```

Or just don't have a base class at all at this point.  If it saves you little code but makes it much more confusing, it is probably not worth it